src/sentry/api/endpoints/project_app_store_connect_credentials.py

```diff
@@ -348,18 +335,19 @@ def post(self, request: Request, project: Project, credentials_id: str) -> Respo
             data["itunesSession"] = session_context.get("itunes_session")
             data["itunesPersonId"] = session_context.get("itunes_person_id")
 
-        symbol_source_config.update(data)
+        new_data = symbol_source_config.to_json()
+        new_data.update(data)
+        symbol_source_config = appconnect.AppStoreConnectConfig.from_json(new_data)
 
-        # We need to have a serialised datetime, but django-rest-framework de-serialised it
-        # into the class.  So, serialize it back.
-        if isinstance(symbol_source_config["itunesCreated"], datetime.datetime):
-            clone = symbol_source_config.copy()
-            clone["itunesCreated"] = symbol_source_config["itunesCreated"].isoformat()
-        else:
-            clone = data
-        jsonschema.validate(clone, APP_STORE_CONNECT_SCHEMA)
+        dsym_download.apply_async(
+            kwargs={
+                "project_id": project.id,
+                "config_id": symbol_source_config.id,
+                "config": symbol_source_config.to_json(),
```

Comment:
As discussed in DMs, it is not recommended to have generic config objects (of variable size) as task arguments, since it might accidentally result in passing KB/MB of data through RabbitMQ. Ideally, we just pass IDs, and then fetch/generate the actual configuration on the consumer side.

Can be fixed in a follow-up PR.

Reply:
this has been fixed now